kivydesigner/uix/kdfilechooser.kv

```diff
@@ -0,0 +1,137 @@
+                source: get_svg_resource('collapse')
+                on_release: root.collapse_all_nodes()
+        ScrollView:
+            id: scrollview
+            do_scroll_x: False
+            Scatter:
+                do_rotation: False
+                do_scale: False
+                do_translation: False
+                size: treeview.size
+                size_hint_y: None
+                KDFileTreeView:
+                    id: treeview
+                    hide_root: True
+                    size_hint_y: None
+                    width: scrollview.width
+                    height: self.minimum_height
+                    on_node_expand: root.controller.entry_subselect(args[1])
+                    on_node_collapse: root.controller.close_subselection(args[1])
+
+<KDFilechooser>:
+    layout: layout
+    dirselect: True
+    KDFilechooserLayout:
+        id: layout
+        controller: root
+        title: self.get_title_name(root.rootpath)
+
+# Use kvlang special syntax '-' to remove the inherited canvas instructions
+<-KDFileChooserEntry>:
+    orientation: 'horizontal'
+    size_hint_y: None
+    spacing: '8dp'
+    height: '48dp' if dp(1) > 1 else '24dp'
+    odd_color: filechooser_background_color
+    even_color: filechooser_background_color
+    color_selected: 0.196, 0.592, 0.992, 0.15
+    canvas.before:
+        Color:
+            # Draw a light blue rectangle if the entry is selected
+            # Disable the selection rectangle if editing the entry
+            # Since the TextInput provides text selection highlighting
+            rgba: self.color_selected if (self.is_selected and not self.in_edit_mode) else filechooser_background_color
+        Rectangle:
+            pos: [self.parent.x, self.y] if self.parent else [0, 0]
+            size: [self.parent.width, self.height] if self.parent else [1, 1]
+        Color:
+            rgba: 1, 1, 1, 1
+        Rectangle:
+            # Populate the entry icon, based on directory and file type
+            source: self.get_entry_icon_path(not self.is_leaf, self.is_open, self.height, self.path)
+            size: self.height / 1.5, self.height / 1.5
+            pos: self.x - dp(20), int(self.center_y - (self.height / 1.5) * .5)
+    canvas.after:
+        Color:
+            rgba: 0.196, 0.592, 0.992, int(self.is_selected)
+        Line:
+            # Draw a darker blue border around selected entries
+            rectangle: [self.parent.x+dp(1), self.y+dp(1), self.parent.width-dp(2), self.height-dp(2)] if self.parent else [0,0,0,0]
+
+# Templates are a deprecated kivy feature, but the implementation of 
+# FileChooserController forces us to use them since the _ENTRY_TEMPLATE
+# class var is used to instantiate new templated entries.
+# The template is initialized with the following context dictionary:
+#  {'name': str, 'get_nice_size': func, 'path': str, 'controller': ref,
+#   'isdir': bool, 'parent': ref, 'sep': str}
```

Comment:
Unfortunately the use of templates causes kivy to issue a deprecated warning. Add an issue to explicitly disable the warning for the `KDFilechooser` since we had no choice but to use templates here.
We should keep the deprecated warning since we should avoid templates when possible.

Reply:
Created issue #18.